docs/core-concepts/app-deployment.md

```diff
@@ -1,17 +1,255 @@
+
+   The resulting files are placed in a directory named `publish` that is in a subdirectory of your project's
+`.\bin\release\netcoreapp1.0` subdirectory.
+
+6. Along with your application's files, the publishing process emits a program database (.pdb) file that contains debugging information about your app. You should not package it with your application's files.
+
+The remaining files can be deployed in any way you'd like. For example, you can package them in a zip file, use a simple `copy` command, or deploy them with any installation package of your choice.
+
+## Deploying a Portable App with Third-Party Dependencies ##
+
+Deploying a portable app with one or more third-party dependencies involves two additional steps before you can run the `dotnet restore` command:
+
+1. Add references to any third-party libraries to the `dependencies` section of your `project.json` file. The following  `dependencies` section uses Json.NET as a third-party library.
+
+
+    ```
+    "dependencies": {
+      "Microsoft.NETCore.App": {
+        "type": "platform",
+        "version": "1.0.0"
+      },
+      "Newtonsoft.Json": "8.0.3"
+    },
+    ```
+
+2. Ensure that you have installed the NuGet package that contains the third-party dependency, since third-party dependencies are resolved from your local NuGet cache.
+
+Note that a portable app with third-party dependencies will only be as portable as its third-party dependencies. For example, if a third-party library only supports OSX, the app will not be portable to Windows systems.
+
+When you deploy your application, any third-party dependencies used in your app are resolved from the local NuGet cache of the system on which your app is running. In other words, the third-party library must be present on the target machine for your app to run successfully.
+
+# Self-Contained Applications #
+
+For a self-contained app, you deploy not only your app and any third-party dependencies, but the version of .NET Core that you build your app with.
+
+## Why Deploy a Self-Contained App? ##
+
+Deploying a self-contained app has two major advantages:
+
+- You have sole control of the version of .NET Core that is deployed with your app. .NET Core can be serviced only by you.
+
+- You can be assured that the target system can run your .NET Core app, since you're providing the version of .NET Core that it will run on.
+
+It also has a number of disadvantages:
+
+- Because .NET Core is included in your deployment package, you must select the target platforms for which you build deployment packages in advance.
+
+- The size of your deployment package is relatively large, since you have to include .NET Core as well as your app and its third-party dependencies.
+
+- Deploying numerous self-contained .NET Core apps to a system can consume significant amounts of disk space, since each app duplicates .NET Core files.
+
+## Deploying a Simple Self-Contained App ## 

```

Comment:
I think the space at the end of this line causes the ending `##` to show in the heading (at least on GitHub).